MAX32655:  Hello_World-riscv Fails to Locate common.ld

This error occurs when I compile the project. How can I solve this problem? Do you know how to solve this error?


d:/maximsdk/tools/gnutools/10.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld.exe: cannot open linker script file common.ld: No such file or directory
collect2.exe: error: ld returned 1 exit status
make[1]: Leaving directory '/d/MaximSDK/Examples/MAX32655/Hello_World-riscv'
make[1]: *** [/D/MaximSDK/Libraries/CMSIS/Device/Maxim/MAX32655/Source/GCC/gcc.mk:351: /d/MaximSDK/Examples/MAX32655/Hello_World-riscv/build/Hello_World-riscv.elf] Error 1
make: *** [Makefile:52: arm] Error 2
"make -r -j 8" terminated with exit code 2. Build might be incomplete.

--- a/Libraries/CMSIS/Device/Maxim/MAX32520/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX32520/Source/GCC/gcc.mk
@@ -92,8 +92,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -224,7 +229,18 @@ endif
 # Add the include file paths to AFLAGS and CFLAGS.
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
+
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX32570/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX32570/Source/GCC/gcc.mk
@@ -95,8 +95,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -276,7 +281,17 @@ endif
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CXXFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX32572/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX32572/Source/GCC/gcc.mk
@@ -95,8 +95,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -259,7 +264,17 @@ endif
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CXXFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX32650/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX32650/Source/GCC/gcc.mk
@@ -92,8 +92,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -261,7 +266,17 @@ endif
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CXXFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX32655/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX32655/Source/GCC/gcc.mk
@@ -95,8 +95,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -261,7 +266,17 @@ endif
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CXXFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX32660/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX32660/Source/GCC/gcc.mk
@@ -92,8 +92,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -261,7 +266,17 @@ endif
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CXXFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX32662/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX32662/Source/GCC/gcc.mk
@@ -92,8 +92,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -261,7 +266,17 @@ endif
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CXXFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX32665/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX32665/Source/GCC/gcc.mk
@@ -95,8 +95,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -251,7 +256,17 @@ endif
 # Add the include file paths to AFLAGS and CFLAGS.
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX32670/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX32670/Source/GCC/gcc.mk
@@ -92,8 +92,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -258,7 +263,17 @@ endif
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CXXFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX32672/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX32672/Source/GCC/gcc.mk
@@ -92,8 +92,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -258,7 +263,17 @@ endif
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CXXFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX32675/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX32675/Source/GCC/gcc.mk
@@ -92,8 +92,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -258,7 +263,14 @@ endif
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CXXFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
-LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
+CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX32680/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX32680/Source/GCC/gcc.mk
@@ -95,8 +95,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -261,7 +266,17 @@ endif
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CXXFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX32690/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX32690/Source/GCC/gcc.mk
@@ -95,8 +95,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -261,7 +266,17 @@ endif
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CXXFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX78000/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX78000/Source/GCC/gcc.mk
@@ -92,8 +92,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -258,7 +263,17 @@ endif
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CXXFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

--- a/Libraries/CMSIS/Device/Maxim/MAX78002/Source/GCC/gcc.mk
+++ b/Libraries/CMSIS/Device/Maxim/MAX78002/Source/GCC/gcc.mk
@@ -92,8 +92,13 @@ endif
 # auto-generated dependencies. Also if this is Cygwin, file paths for ARM GCC
 # will be converted from /cygdrive/c to C:.
 ################################################################################
-ifneq ($(findstring CYGWIN, ${shell uname -s}), )
+UNAME := $(shell uname -s)
+ifneq ($(findstring CYGWIN, $(UNAME)), )
 CYGWIN=True
+endif
+
+ifneq ($(findstring MSYS, $(UNAME)), )
+MSYS=True
 endif
 
 # Get the prefix for the tools to use.
@@ -258,7 +263,17 @@ endif
 AFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
 CXXFLAGS+=${patsubst %,-I%,$(call fixpath,$(IPATH))}
+ifneq ($(MSYS),)
+# 2-27-2023:  This workaround was added to resolve a linker bug introduced
+# when we started using ln_args.txt.  The GCC linker expects C:/-like paths
+# on Windows if arguments are passed in from a text file.  However, ln_args
+# is parsed through a regex that misses the edge case -L/C/Path/... because
+# of the leading "-L".  We use cygpath here to handle that edge case before
+# parsing ln_args.txt.
+LDFLAGS+=${patsubst %,-L%,$(shell cygpath -m $(LIBPATH))}
+else
 LDFLAGS+=${patsubst %,-L%,$(call fixpath,$(LIBPATH))}
+endif
 
 ################################################################################
 # The rule for building the object file from each C source file.

